src/main/scala/diplomacy/Resources.scala

```diff
@@ -392,6 +394,10 @@ object BindingScope
     case x: BindingScope => find(x.parent).orElse(Some(x))
     case x => find(x.parent)
   }
+
+  var bindingScopes = new collection.mutable.ArrayBuffer[BindingScope]()
+
+  def add(bs: BindingScope) = BindingScope.bindingScopes.+=:(bs)
```

Comment:
Minor nit, but this could use infix notation
```suggestion
  def add(bs: BindingScope) = bs +=: BindingScope.bindingScopes
```

You could alternatively explicitly use `prepend`, I think either is better the `.+=:`